enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,366 @@
+of 10% rather than absolute value of 1, on a cluster of 250 nodes that generally
+reduced rollout time from 80 minutes to 10 minutes. The multus, sdn, ovs DaemonSets
+have not been updated but likely could be now that workload critical processes
+have been moved to the host. Before doing this given the risk we should devise
+a method to canary updates to these DaemonSets. Any other options to speed up
+updates to these DaemonSets which are deployed to all Linux nodes should also be
+considered.
+
+Once these new deployment patterns have been vetted in 4.8 or 4.9 these changes
+should be backported through to 4.6.
+
+#### As an admin, my workloads within a failure domain are rescheduled a minimum number of times
+
+Currently the MCO cycles nodes in a deterministic but unintentional manner. The
+workload on that host which can be relocated scatters to other hosts according
+to scheduling rules. This has been measured and shown that approximately 70% of
+all reschedulable pods are rescheduled more than necessary if they were
+rescheduled in an optimal manner. Optimal rescheduling should be closer to (1 /
+node_count_per_failure_domain) percentage of pods being rescheduled more than
+once during a rolling restart of a given failure domain.
+
+We should make the MCO topology aware so that it upgrades all of the host in one
+failure domain because anti-affinity scheduling is used in most scenarios. We
+should provide some mechanism to rank upgraded nodes as slightly higher priority
+in scheduler configuration. Perhaps a timestamp of configuration generation that
+the scheduler then biases toward nodes with the greatest timestamps as a last in
+chain priority.
+
+We should consider enabling the rescheduler as well to ensure that when the
+upgrade completes we rebalance so that the last node ends up with proportionate
+workload.
+
+#### As an admin of a disconnected cluster, I have a tool which helps me mirror all content necessary for EUS to EUS upgrades
+
+Assume that 4.6.eus-z -> 4.7.z -> 4.8.z -> 4.9.z -> 4.10.z are all in the eus-4.10
+channel providing a chain of upgrades that eventually lead to 4.10. We should build
+a graph viewer / solver somewhere which allows an admin of a 4.6 cluster hoping
+to upgrade to 4.10 to discover all content which must be mirrored between their
+current version and their eventual 4.10 target.
+
+In general the graph viewer should provide a complete path from the cluster's
+current version to the latest .z in the selected channel assuming such a path exists.
+If someone were to provide cluster version 4.4.3 but select the stable-4.6 channel
+no such path exists and it should indicate that, if they were to select stable-4.5
+it would provide the upgrade path of 4.4.3 to 4.4.29 to 4.5.24 which is the latest 4.5.z.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The main caveat of what's outlined here is that it does not reduce reboots
```

Comment:
I think we need to more precise about the desires to eliminate reboots. In various conversations I have had, we seem to be conflating workload interruption with reboots. If we are talking about workload interrupts then the solutions that we come up might look _very_ different. For example, live-migration for workload interruption over reboot which might be a _capacity_ problem.

Reply:
I'll write up a separate enhancement that focuses on removal of reboots and we can propose alternatives there knowing that it's a mix of both concern over actual reboot and workload disruption due to rescheduling.